var/spack/repos/builtin/packages/hdf5/package.py

```diff
@@ -156,6 +156,20 @@ def configure_args(self):
 
         return ["--with-zlib=%s" % spec['zlib'].prefix] + extra_args
 
+    def configure(self, spec, prefix):
+        # Run the default autotools package configure
+        super(Hdf5, self).configure(spec, prefix)
+
+        if '@:1.8.14' in spec:
```

Comment:
> The decorator is nice for sanity checks because you can separate them out by functions, but it is not all that readable...

I think it is nice also for tweaks like this one, because of the same reasons: you can separate different tweaks into different functions. Honeslty I see no need for a monolithic `post_<phase>` method.

Totally agree instead on making it a free function decorator. It cleans the syntax by a great deal.